tests/hotplug/cpu.go

```diff
@@ -0,0 +1,194 @@
+			enabled  int
+			disabled int
+		}
+		countDomCPUs := func(vmi *v1.VirtualMachineInstance) (count cpuCount) {
+			domSpec, err := tests.GetRunningVMIDomainSpec(vmi)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(domSpec.VCPUs).NotTo(BeNil())
+			for _, cpu := range domSpec.VCPUs.VCPU {
+				if cpu.Enabled == "yes" {
+					count.enabled++
+				} else {
+					ExpectWithOffset(1, cpu.Enabled).To(Equal("no"))
+					ExpectWithOffset(1, cpu.Hotpluggable).To(Equal("yes"))
+					count.disabled++
+				}
+			}
+			return
+		}
+		It("should have spare CPUs that be enabled and more pod CPU resource after a migration", func() {
+			By("Creating a running VM with 1 socket and 2 max sockets")
+			const (
+				maxSockets uint32 = 2
+			)
+
+			vmi := libvmi.NewAlpine()
+			vmi.Spec.Domain.CPU = &v1.CPU{
+				Cores:   2,
+				Sockets: 1,
+				Threads: 1,
+			}
+			vm := tests.NewRandomVirtualMachine(vmi, true)
+			vm.Spec.LiveUpdateFeatures = &v1.LiveUpdateFeatures{
+				CPU: &v1.LiveUpdateCPU{
+					MaxSockets: pointer.P(maxSockets),
+				},
+			}
+
+			vm, err := virtClient.VirtualMachine(vm.Namespace).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(func() error {
+				vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+				return err
+			}).ShouldNot(HaveOccurred())
+			libwait.WaitForSuccessfulVMIStart(vmi)
+
+			By("Ensuring the computee container has 200m CPU")
+			pod := tests.GetVmiPod(virtClient, vmi)
+			var compute *k8sv1.Container
+			for _, container := range pod.Spec.Containers {
+				if container.Name == "compute" {
+					compute = &container
+					break
+				}
+			}
+			Expect(compute).NotTo(BeNil(), "failed to find compute container")
+			Expect(*compute.Resources.Requests.Cpu()).To(Equal(resource.MustParse("200m")))
+
+			By("Ensuring the libvirt domain has 2 enabled cores and 2 hotpuggable cores")
+			Expect(countDomCPUs(vmi)).To(Equal(cpuCount{
+				enabled:  2,
+				disabled: 2,
+			}))
+
+			By("Enabling the second socket")
+			patchData, err := patch.GenerateTestReplacePatch("/spec/template/spec/domain/cpu/sockets", 1, 2)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, patchData, &k8smetav1.PatchOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			var migration *v1.VirtualMachineInstanceMigration
+
+			By("Ensuring live-migration started")
+			Eventually(func() bool {
+				migrations, err := virtClient.VirtualMachineInstanceMigration(vm.Namespace).List(&k8smetav1.ListOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				for _, mig := range migrations.Items {
+					if mig.Spec.VMIName == vmi.Name {
+						migration = mig.DeepCopy()
+						return true
+					}
+				}
+				return false
+			}, 30*time.Second, time.Second).Should(BeTrue())
+			tests.ExpectMigrationSuccess(virtClient, migration, tests.MigrationWaitTime)
```

Comment:
What happens if the migration ends really fast?
Would this still capture it in time? Is the vmim object still around after migration is finished?